lib/matplotlib/tests/test_datetime.py

```diff
@@ -89,8 +89,17 @@ def test_barh(self):
     @pytest.mark.xfail(reason="Test for boxplot not written yet")
     @mpl.style.context("default")
     def test_boxplot(self):
-        fig, ax = plt.subplots()
-        ax.boxplot(...)
+        data = [datetime.datetime(2023, 10, 5, 18, 52, 59),
+                datetime.datetime(2023, 10, 4, 18, 53, 0),
+                datetime.datetime(2023, 10, 3, 18, 53, 1),
+                datetime.datetime(2023, 10, 2, 18, 53, 2),
+                datetime.datetime(2023, 10, 1, 18, 53, 3)]
+        datums = np.array([datetime.datetime.toordinal(d) for d in data])
+        plt.boxplot(datums, labels=['my data'], showfliers=True)
```

Comment:
```suggestion
        plt.boxplot(data, labels=['my data'], showfliers=True)
```

We actually very specifically _want_ the datetimes passed into the plotting methods here. as it stands, this test is not actually testing the unit conversion at all.